### PR TITLE
Squash #261: Use new H3MemoryManager system

### PR DESCRIPTION
**Pull request by ajfriend**
Co-authored-by: AJ Friend <ajfriend@gmail.com>

### CODE REVIEW

File: src/h3/_cy/edges.pyx [hunk 1]
```diff
@@ -67,13 +64,9 @@ cpdef H3int[:] edges_from_cell(H3int origin):
...
 
-    ptr = create_ptr(6)
-
-    check_for_error(
-        h3lib.originToDirectedEdges(origin, ptr)
-    )
-
-    mv = create_mv(ptr, 6)
+    hmm = H3MemoryManager(6)
+    h3lib.originToDirectedEdges(origin, hmm.ptr)
```
Comment (isaacbrodsky) [inline src/h3/_cy/edges.pyx:68]: check_for_error gets dropped here?

File: src/h3/_cy/memory.pyx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    mv.callback_free_data = h3_free
+
+    # responsibility for the memory moves from this object to the array/memoryview
+    x.ptr = NULL
+    x.n = 0
+
+    return mv
+
+
+cdef class H3MemoryManager:
```
Comment (isaacbrodsky) [inline src/h3/_cy/memory.pyx:141]: Please add a comment for this class

File: src/h3/_cy/memory.pyx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    mv.callback_free_data = h3_free
+
+    return mv
+
+
+cpdef H3int[:] iter_to_mv(hexes):
+    """ hexes needs to be an iterable that knows its size...
+    or should we have it match the np.fromiter function, which infers if not available?
+    """
+    # cdef array x  # this needs to be commented out to avoid an error
```
Comment (isaacbrodsky) [inline src/h3/_cy/memory.pyx:203]: delete unused line? is the comment semantically relevant?

File: src/h3/_cy/edges.pyx (hunk 1 above)
Comment (ajfriend) [reply src/h3/_cy/edges.pyx]: good catch!

**Review by isaacbrodsky — APPROVED** [review]

**Comment by ajfriend** [comment]
I'm currently confused around this: https://groups.google.com/g/cython-users/c/mlIs8xnQDoY

File: src/h3/_cy/memory.pyx (hunk 3 above)
Comment (ajfriend) [reply src/h3/_cy/memory.pyx]: Figured out what was going on. removed.

File: src/h3/_cy/memory.pyx (hunk 2 above)
Comment (ajfriend) [reply src/h3/_cy/memory.pyx:162]: done.

**Pull request merged**